### PR TITLE
Synced some big endian code to mirror little endian branches

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

**Comment by seanm** [comment]
@malaterre here I'm bringing back some code you didn't merge last time (not sure why).  The idea here is to make code in the big endian branches match the little endian branches.

CC @issakomi

File: Source/Common/gdcmByteSwap.txx [hunk 1]
```diff
@@ -106,7 +106,7 @@ void Swap4(T &a, SwapCode const &swapcode)
 {
 #ifndef GDCM_WORDS_BIGENDIAN
   if ( swapcode == gdcm::SwapCode::BigEndian || swapcode == gdcm::SwapCode::BadBigEndian )
-    a = (T)(( a << 8 ) | ( a >> 8 ));
+    a = ( a << 8 ) | ( a >> 8 );
```
Comment (malaterre) [inline Source/Common/gdcmByteSwap.txx:109]: could you include the warning/error reported here ? Here is what I see using git blame: 1033e91199

File: Source/Common/gdcmSwapper.txx [hunk 2]
```diff
@@ -135,7 +135,7 @@ namespace gdcm
 
   template <> inline Tag SwapperNoOp::Swap<Tag>(Tag val)
     {
-    return Tag( Swap(val.GetGroup()), Swap(val.GetElement()) );
+    return Tag( Swap((uint32_t)val.GetElementTag()) );
```
Comment (malaterre) [inline Source/Common/gdcmSwapper.txx:138]: Same comment, refer to 874fa717ac

File: Source/Common/gdcmByteSwap.txx (hunk 1 above)
Comment (seanm) [reply Source/Common/gdcmByteSwap.txx:109]: There is no warning/error. This patch comes from me comparing the code inside `GDCM_WORDS_BIGENDIAN` versus the code outside it.  ex, in master, why are these two lines different?

https://github.com/malaterre/GDCM/blob/008e437d6ea81979bfa23301ae6e62f82272fe64/Source/Common/gdcmByteSwap.txx#L109

https://github.com/malaterre/GDCM/blob/008e437d6ea81979bfa23301ae6e62f82272fe64/Source/Common/gdcmByteSwap.txx#L112

File: Source/Common/gdcmSwapper.txx (hunk 2 above)
Comment (seanm) [reply Source/Common/gdcmSwapper.txx:138]: Same. Compare:

https://github.com/malaterre/GDCM/blob/008e437d6ea81979bfa23301ae6e62f82272fe64/Source/Common/gdcmSwapper.txx#L214

https://github.com/malaterre/GDCM/blob/008e437d6ea81979bfa23301ae6e62f82272fe64/Source/Common/gdcmSwapper.txx#L138

Why are they different?

Comment (seanm) [reply Source/Common/gdcmSwapper.txx:138]: So in fact in 874fa717ac769ecf7a43c189a7974b1eb100aade looks like you changed only the big endian branch, and not the little endian branch.

File: Source/Common/gdcmByteSwap.txx (hunk 1 above)
Comment (seanm) [reply Source/Common/gdcmByteSwap.txx:109]: Do you see what I mean now?

**Pull request merged**